From b10ccbc58d8d7f2ebb9ded079ec2a36e2f4603ad Mon Sep 17 00:00:00 2001
From: Tomas Bzatek <tbzatek@redhat.com>
Date: Thu, 27 Jan 2011 10:22:10 +0000
Subject: Prevent a crash in nautilus_file_peek_display_name() on invalid NautilusFile

This is more a workaround only, expect assert failures at other
places when something bad happens. There's a race condition somewhere,
this patch only prevents immediate crash.

Patch by Marcus Husar <marcus.husar@rose.uni-heidelberg.de>

https://bugzilla.gnome.org/show_bug.cgi?id=602500
---
diff --git a/libnautilus-private/nautilus-file.c b/libnautilus-private/nautilus-file.c
index 2c5b868..c854163 100644
--- a/libnautilus-private/nautilus-file.c
+++ b/libnautilus-private/nautilus-file.c
@@ -3780,6 +3780,9 @@ nautilus_file_peek_display_name (NautilusFile *file)
 	const char *name;
 	char *escaped_name;
 
+	if (file == NULL || nautilus_file_is_gone (file))
+		return "";
+
 	/* Default to display name based on filename if its not set yet */
 	
 	if (file->details->display_name == NULL) {
--
cgit v0.9
